File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/config/AWSRegionClientConfiguration.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0.
 */

package software.amazon.smithy.aws.swift.codegen.config

import software.amazon.smithy.aws.swift.codegen.AWSSwiftDependency
import software.amazon.smithy.codegen.core.Symbol
import software.amazon.smithy.swift.codegen.SwiftTypes
import software.amazon.smithy.swift.codegen.config.ClientConfiguration
import software.amazon.smithy.swift.codegen.config.ClientConfiguration.Companion.runtimeSymbol
import software.amazon.smithy.swift.codegen.config.ConfigProperty
import software.amazon.smithy.swift.codegen.model.toNullable

class AWSRegionClientConfiguration : ClientConfiguration {
    override val swiftProtocolName: Symbol
        get() = runtimeSymbol("AWSRegionClientConfiguration", AWSSwiftDependency.AWS_CLIENT_RUNTIME)

    override val properties: Set<ConfigProperty>
        get() = setOf(
            ConfigProperty("region", SwiftTypes.String.toNullable(), "AWSClientConfigDefaultsProvider.region(region)", true, true),
            ConfigProperty("signingRegion", SwiftTypes.String.toNullable(), "AWSClientConfigDefaultsProvider.region(region)", true, true)
        )
}

File: Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Config/AWSRegionClientConfiguration.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

/// Protocol for AWS region client configuration.
///
/// Note: This protocol will inherit `Sendable` after the deprecation period of class-based configurations.
/// The struct-based configuration already conforms to `Sendable` via `@unchecked Sendable`.
public protocol AWSRegionClientConfiguration {

    /// The AWS region to use, i.e. `us-east-1` or `us-west-2`, etc.
    ///
    /// If no region is specified here, one must be specified in the `~/.aws/configuration` file.
    var region: String? { get set }

    /// The signing region to be used for signing AWS requests.
    ///
    /// If none is specified, it is supplied by the SDK.
    var signingRegion: String? { get set }
}
